utils/makefile.sphinx

```diff
@@ -172,20 +172,19 @@ PHONY += prepare-gh-pages
 prepare-gh-pages:
 	cp -r $(DOCS_DIST)/* $(GH_PAGES)/
 	touch $(GH_PAGES)/.nojekyll
-	echo "<html><head><META http-equiv='refresh' content='0;URL=index.html'></head></html>" > $(GH_PAGES)/404.html	
+	echo "<html><head><META http-equiv='refresh' content='0;URL=index.html'></head></html>" > $(GH_PAGES)/404.html
 
 PHONY += $(GH_PAGES)
 $(GH_PAGES)::
 	$(MAKE) docs
-	[ -d "gh-pages/.git" ] || git clone $(GIT_URL) gh-pages
-	-cd $(GH_PAGES); git checkout gh-pages >/dev/null
-	-cd $(GH_PAGES); git pull
-	-cd $(GH_PAGES); ls -A | grep -v '.git$$' | xargs rm -rf
+	. ./utils/brand.env; [ -d "gh-pages/.git" ] || git clone $$GIT_URL gh-pages
+	-cd $(GH_PAGES); git checkout --orphan gh-pages || exit 0
```

Comment:
Does it make sense to replace `git clone $$GIT_URL gh-pages` by `git clone $$(git config --get remote.origin.url)` ?

A loud and clear fail rather than `|| exit 0` : with a soft fail, we won't know if the documentation is deployed without looking at the logs.

Reply:
> replace git clone $$GIT_URL gh-pages by git clone 

makes sense, i will give it a try

> A loud and clear fail rather than || exit 0 : with a soft 

This catches only the checkout if branch is already checked out. Remember, this is a Makefile, this exit 0 does only affect this one line .. the following lines will proceed

Reply:
> Remember, this is a Makefile, this exit 0 does only affect this one line .

I forgot that point. Thank you.
The thing that is not clear, is when this line fails ?

Reply:
It fails when you run this target second time.  Here is a example what is suppressed by the exit 0:

     make gh-pages
     ....
     make gh-pages
     ...
     cd gh-pages; git checkout --orphan gh-pages
     fatal: A branch named 'gh-pages' already exists.
     make: [utils/makefile.sphinx:181: gh-pages] Error 128 (ignored)
     rm -rf gh-pages/* gh-pages/.gitignore
     make prepare-gh-pages
     ....

And this shows how it looks like with the exit 0

     ...
     cd gh-pages; git checkout --orphan gh-pages || exit 0
     fatal: A branch named 'gh-pages' already exists.
     rm -rf gh-pages/* gh-pages/.gitignore
     make prepare-gh-pages
     ....

Reply:
Reading this https://stackoverflow.com/questions/21151178/shell-script-to-check-if-specified-git-branch-exists

Would
`-cd $(GH_PAGES); [ $$(git branch --list gh-pages) ] || git checkout --orphan gh-pages`
do the trick ?

Note: if for some reason the clone already exists, and the current branch is not gh-pages, then the script fails.

Reply:
Before you investigate more time .. I will try to implement a different solutions using git worktree .. which will make cloning from a remote obsolete .. anyway thanks for your comments which helps me to take a closer look at this implementation  and to question the concepts I chose.

Reply:
> Would .. do the trick ?

Yes, but what do we win by this except additional git calls?  I will try to give a short explanation.. 

The `exit 0` exists only to suppress error messages from 'make' itself.  In the front there is a minus:

    -cd $(GH_PAGES); git checkout --orphan gh-pages || exit 0

This minus instructs [make to go further even if exit status of this line is not zero](https://www.gnu.org/software/make/manual/make.html#Errors) but also to report if this shell command fails. Doing both (exit 0 and the minus) might be superfluous (at least the minus), but the minus here is for better readability saying: _whatever the exit code of this line, ignore it and go further_

Reply:
I'm not very familiar with Makefile.

So my suggestion is `cd $(GH_PAGES); [ $$(git branch --list gh-pages) ] || git checkout --orphan gh-pages`  ( I feel `|| exit 0` is like a `try: ... except: pass` which may hide an unexpected exceptions).

But that's okay with `-cd $(GH_PAGES); git checkout --orphan gh-pages || exit 0`.